source/extensions/tracers/opentelemetry/tracer.cc

```diff
@@ -118,14 +113,20 @@ void Tracer::flushSpans() {
   ExportTraceServiceRequest request;
   // A request consists of ResourceSpans.
   ::opentelemetry::proto::trace::v1::ResourceSpans* resource_span = request.add_resource_spans();
-  opentelemetry::proto::common::v1::KeyValue key_value =
-      opentelemetry::proto::common::v1::KeyValue();
-  opentelemetry::proto::common::v1::AnyValue value_proto =
-      opentelemetry::proto::common::v1::AnyValue();
-  value_proto.set_string_value(std::string{service_name_});
-  key_value.set_key(std::string{kServiceNameKey});
-  *key_value.mutable_value() = value_proto;
-  (*resource_span->mutable_resource()->add_attributes()) = key_value;
+  resource_span->set_schema_url(resource_.schemaUrl);
+
+  // add resource attributes
+  for (auto const& att : resource_.attributes) {
```

Comment:
I think this makes a const reference to an object that would subsequently be modifyable. What we want is a reference (and I don't think it makes a difference if that one is const) to a const object, so maybe:
```suggestion
  for (const auto const& att : resource_.attributes) {
```
or
```suggestion
  for (const auto & att : resource_.attributes) {
```
would also work? not sure right now

Reply:
Unless the values in attributes are already `const` in which case the compiler should expand the auto to `const attribute_t` anyway. That's the problem with auto and no IDE - I don't know what the type of this actually is.

Reply:
@samohte this is correct right? `.attributes` is a Map<string,string>, so `att` here is a `pair`. by `const&` then we say we want the pair to be constant and a reference to avoid copying. Or am I wrong?

Reply:
`const auto &` and `auto const &` are equal, see https://isocpp.org/blog/2018/09/quick-q-is-auto-const-and-const-auto-the-same
C++ does not allow to rebind a reference.